include/z64audio.h

```diff
@@ -125,7 +125,7 @@ typedef struct {
     /* 0x00 */ u32 medium : 2;
     /* 0x00 */ u32 unk_bit26 : 1;
     /* 0x00 */ u32 unk_bit25 : 1;
-    /* 0x01 */ u32 size : 24;
+    /* 0x01 */ size_t size : 24;
```

Comment:
According to the C89 standard, the only valid types for a bit-field are `int`, `unsigned int` and `signed int`.
https://port70.net/~nsz/c/c89/c89-draft.html#3.5.2.1
C99 extended this to allow `_Bool` types too.
https://port70.net/~nsz/c/c99/n1256.html#6.7.2.1

(in case you don't like reading the standard: https://en.cppreference.com/w/c/language/bit_field)

The use of any other type besides the listed above are compiler extensions, which I think we should try to keep to a minimum.

So this should be reverted back to `u32`.

Reply:
With IDO `u32 == size_t` and I will update our `size_t` to be an unsigned int (which is what MSVC does IDK about GCC and IDO) if thats not OK then we need to change all the bit-fields to all use `unsigned int`

Reply:
You are missing the point.

Everybody knows IDO is meant to be equivalent to a `u32`, which is just an `unsigned int`. The problem is this change would not be portable to other compilers which may not have those extensions.

Now that you mentioned MSVC, let's see what does its docs say about `size_t` as an example:
> `size_t` (`unsigned __int64` or `unsigned integer`, depending on the target platform)

https://docs.microsoft.com/en-us/cpp/c-runtime-library/standard-types?view=msvc-170

So, if MSVC tried to build this struct targeting a 64-bits machine, then `size_t` would expand to `unsigned __int64`, which would be violating the standard.

In the other hand, let's take a look at MSVC's docs for bit-fields:
```
struct-declarator:
    declarator
    type-specifier declaratoropt : constant-expression
```
The _constant-expression_ specifies the width of the field in bits. The _type-specifier_ for the `declarator` must be `unsigned int`, `signed int`, or `int`, and the _constant-expression_ must be a nonnegative integer value. 

https://docs.microsoft.com/en-us/cpp/c-language/c-bit-fields?view=msvc-170#syntax

Even Microsoft says the valid types for bit-fields should be `unsigned int`, `signed int` or `int`.

Scrolling to the "Microsoft Specific" section, it says a Microsoft extension to the standard allows using `char`, `long` and `short` (either `signed` or `unsigned`), but it doesn't mention anything about `unsigned __int64`, which is what the `size_t` type expands on 64-bits targets according to their docs. 

So no idea what would happen when using `size_t` as the type of a bit-field on MSVC.  

And this is just using MSVC as an example. Other compilers may have a different behaviour or not implement this extension at all. This is why we should avoid using compiler extensions when we can avoid them.